### PR TITLE
Pharo8.0-ffi

### PR DESCRIPTION
**Pull request by zecke**
Add testcase for the internal structure of the read FFIExternalStructure

Any idea how to make this more than a change detector? I want to asses that "a" is a valid object and can be used for future FFI. I tried to call another callback from within the callback but that seemed to deadlock.

### CODE REVIEW

**Comment by estebanlm** [comment]
yes, that's more or less how I was expecting it to work :)
I mean: no aliens in the side.

File: src/FreeType/ExternalAddress.extension.st [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Extension { #name : #ExternalAddress }
+
+{ #category : #'*FreeType' }
```
Comment (tesonep) [inline src/FreeType/ExternalAddress.extension.st]: Should be moved to UnifiedFFI package

**Review by tesonep — CHANGES_REQUESTED** [review]

File: src/FreeType/ExternalAddress.extension.st (hunk 1 above)
Comment (zecke) [reply src/FreeType/ExternalAddress.extension.st]: FFIExternalStructureReferenceHandle>>#asExternalAddress is inside the FreeType package (https://github.com/pharo-project/pharo/blob/Pharo8.0/src/FreeType/FFIExternalStructureReferenceHandle.extension.st). Should both move to UnifiedFFI?

Comment (guillep) [reply src/FreeType/ExternalAddress.extension.st]: Could be nice...

**Comment by zecke** [comment]
ping to prevent auto close.

**Comment by zecke** [comment]
keep open for now.

File: src/FreeType/ExternalAddress.extension.st (hunk 1 above)
Comment (estebanlm) [reply src/FreeType/ExternalAddress.extension.st]: Hi, @zecke 
can you perform the recategorisation, so I can integrate this?
Thanks!
Esteban

Comment (zecke) [reply src/FreeType/ExternalAddress.extension.st]: Done. I moved both into UnifiedFFI

**Comment by MarcusDenker** [comment]
I it seems now the image crashes when we use iceberg (directly on cloning a project). This might be a side effect of this PR?

**Comment by estebanlm** [comment]
yes. We need to revert.

**Comment by estebanlm** [comment]
@zecke somehow this change crashes the cloning of projects using HTTPS (maybe other cases too). 
Probably an invalid #free call somewhere, but it needs to be dug. 

Here the way to reproduce the crash (provided by @astares): 

- Open Iceberg
- click on "Add"
- select "Clone from GitHub"
- use "astares" as Owner name and "pharo" as Project name, select "HTTPS" as Protocol, click on "Accept"

Pharo tries to clone the repo - but then the Pharo process just dies.

On Ubuntu the consol tells:

 free(): invalid pointer
 ./pharo-ui: line 11:  3604 Aborted                 (core dumped) "$DIR"/"pharo-vm/pharo" "$@"

**Comment by jecisc** [comment]
A fetch is enough to crash the image for me.

**Comment by astares** [comment]
Details described here: https://github.com/pharo-project/pharo/issues/3461

**Pull request merged**